Core/tests/Unit/RequestWrapperTest.php

```diff
@@ -668,6 +669,81 @@ public function testPassingInRetryOptions()
         $this->assertTrue($retryFunctionCalled);
         $this->assertTrue($retryListenerCalled);
     }
+
+    /**
+     * Test to verify auth related headers fetching and updation if
+     * credential fetcher implements UpdateMetadataInterface too, else
+     * going ahead with access_code fetching followed by auth header
+     * creation and updation.
+     */
+    public function testFetchingCredentialAsAuthHeader()
+    {
+        $credentialsFetcher = $this->prophesize(FetchAuthTokenInterface::class);
+        $credentialsFetcher->willImplement(UpdateMetadataInterface::class);
+
+        $accessToken = 'abc';
+        $credentialsFetcher->updateMetadata(Argument::cetera())->willReturn([
+            'authorization' => ['Bearer ' . $accessToken]
+        ]);
+
+        // We have to mock this message because RequestWrapper wraps the credentials using the
+        // FetchAuthTokenCache class
+        $credentialsFetcher->getCacheKey()
+            ->willReturn(null);
+
+        // We have to mock this message because FetchAuthTokenCache class' updateMetadata()
+        // internally calls getLastReceivedToken() as a part of token fetching.
+        $credentialsFetcher->getLastReceivedToken()->willReturn(null);
+
+        $requestWrapper = new RequestWrapper([
+            'credentialsFetcher' => $credentialsFetcher->reveal(),
+            'httpHandler' => function ($request, $options = []) use ($accessToken) {
+                $authHeader = $request->getHeaderLine('authorization');
+                $this->assertEquals('Bearer ' . $accessToken, $authHeader);
+                return new Response(200);
+            }
+        ]);
+
+        $requestWrapper->send(
+            new Request('GET', 'http://www.example.com')
+        );
+    }
+
+    public function testFetchingCredentialAsAuthHeaderWithOverlappingHeaders()
+    {
+        $credentialsFetcher = $this->prophesize(FetchAuthTokenInterface::class);
+        $credentialsFetcher->willImplement(UpdateMetadataInterface::class);
+
+        $accessToken = 'abc';
+        $credentialsFetcher->updateMetadata(Argument::that(function ($arg) {
+            return array_key_exists('x-goog-api-client', $arg);
+        }), Argument::cetera())->willReturn([
+            'authorization' => ['Bearer ' . $accessToken],
+            'x-goog-api-client' => ['xyz']
+        ]);
```

Comment:
This needs a `shouldBeCalled` so that we know for sure it was invoked as expected
```suggestion
        ])->shouldBeCalledOnce();
```

Reply:
Makes sense, fixed.